src/GraphQL/Types/Schema.cs

```diff
@@ -249,6 +249,37 @@ public void RegisterTypes(params Type[] types)
             }
         }
 
+        private List<(Type clrType, Type graphType)> _clrToGraphTypeMappings;
+
+        /// <inheritdoc/>
+        public void RegisterTypeMapping(Type clrType, Type graphType)
+        {
+            (_clrToGraphTypeMappings ??= new List<(Type, Type)>()).Add((clrType ?? throw new ArgumentNullException(nameof(clrType)), graphType ?? throw new ArgumentNullException(nameof(graphType))));
+        }
+
+        /// <inheritdoc/>
+        public IEnumerable<(Type clrType, Type graphType)> TypeMappings
+        {
+            get
+            {
+                if (_clrToGraphTypeMappings != null)
+                {
+                    foreach (var (clrType, graphType) in _clrToGraphTypeMappings)
+                        yield return (clrType, graphType);
+                }
```

Comment:
Does this work?

```suggestion
                if (_clrToGraphTypeMappings != null)
                {
                    return _clrToGraphTypeMappings;
                }
                else
                {
                    return Enumerable.Empty<(Type, Type)>();
                }
```

Reply:
Keep in mind that this change allows optimizations by LINQ methods for any LINQ queries executed against the returned IEnumerable.

Reply:
OK